cluster/swarm/cluster.go

```diff
@@ -170,6 +170,11 @@ func (c *Cluster) createContainer(config *cluster.ContainerConfig, name string,
 		config.SetSwarmID(swarmID)
 	}
 
+	if network := c.Networks().Get(config.HostConfig.NetworkMode); network != nil {
+		config.AddConstraint("node==~" + network.Engine.Name)
```

Comment:
I'm not sure if this added constraint is helpful or not. If a container is rescheduled 3 times at 3 different nodes, would all these soft constraints added up to `node==~node1; node==~node2; node==~node3`?